### ISSUE TITLE
services.service_name.env_file.0 must be a string

### ISSUE THREAD

**Issue by mihalt** [hoverkraft-tech/compose-action#54]
I am trying to do something like this 
```
name: Django CI

on:
  push:
    branches: [ "production", "developing" ]
  pull_request:
    branches: [ "production", "developing" ]
  workflow_dispatch:

env:
  REGISTRY: ghcr.io
  IMAGE_NAME: ${{ github.repository }}

jobs:
  testing:
    runs-on: ubuntu-latest

    permissions:
      contents: read
      packages: write

    steps:
      - uses: actions/checkout@v4
      - uses: hoverkraft-tech/compose-action@v1.5.1
      - name: testing
        run: "docker compose exec app pytest"

      - name: Log in to the Container registry
        uses: docker/login-action@v3
        with:
          registry: ${{ env.REGISTRY }}
          username: ${{ github.actor }}
          password: ${{ secrets.GITHUB_TOKEN }}

      - name: Extract metadata (tags, labels) for Docker
        id: meta
        uses: docker/metadata-action@v5
        with:
          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}

      - name: Build and push Docker image
        uses: docker/build-push-action@f2a1d5e99d037542a71f64918e516c093c6f3fc4
        with:
          context: .
          push: true
          tags: ${{ steps.meta.outputs.tags }}
          labels: ${{ steps.meta.outputs.labels }}
```

And I get an error like `Error: compose up failed {"exitCode":15,"err":"validating /home/runner/work/.../docker-compose.yml: services.app.env_file.0 must be a string\n","out":""}` and `Error: compose down failed [object Object]`

In `docker-compose.yml` this service strings looks like:
```
  app:
    env_file:
      - path: ".env"
        required: true
      - path: "overwrite.env"
        required: false
```

**Comment by mihalt** [comment]
> @mihalt the `required` attribute is available with Docker Compose version 2.24.0 or later. (https://docs.docker.com/compose/compose-file/05-services/#env_file). From what I now Github runners are using docker compose`2.23.3`

so, what should I do in this case? manually update docker compose on github runner by separate `apt install` command?

**Comment by neilime** [comment]
Yes this is a workaround, the best thing will be that the action provides a way to ensure a specific version.
I'm creating a issue to track this feature.
